stack/framework/hal/platforms/NATIVE/README.md

```diff
@@ -0,0 +1,43 @@
+# Overview
+
+The native platform allows the stack to be run on a Linux computer.  The
+emulation presently supports:
+
+- Push button emulation with 10 virtual buttons using numbered keys 0-9
+- Unlimited mumber of sensor application instances
+- Console serial output to the same terminal the application is launched
+- Emulated radio based on multi-cast IP communications
+- Emulated UART in conjunction with the socat tool to allow pyd7a domain
+  to interact with the gateway using the serial modem interface
+- Option to connect to real peripherals (eg GPS) using a real UART
+  on your Linux machine via the emulated UART layer
+
+# Why is this useful?
+
+- Access to physical hardware is not always possible
+- Allow application and stack to be tested independently of hardware
+- Full stack debug trace can be CPU intensive on small embedded CPUs
+- More rapidly try out stack or application ideas
+
+# Quick start
+
+Launch virtual serial ports using socat as follows:
+
+socat pty,link=/tmp/S0,raw,echo=0 pty,link=/tmp/S1,raw,echo=0 &
+
+Run the following command to build using the assigned device /tmp/S0 from socat above:
```

Comment:
Right now the fd to use for the UART is a build time parameter, which means that on one host PC it is only possible to run one node which uses UART right? Unless you have multiple build directories but this is cumbersome. I was wondering if it would not be easier and more flexible to pass the filename to use as a parameter to the binary (eg `./modem -d /tmp/S1`)? This would simplify the build a bit and allow to start multiple nodes using different serial ports easily.

This is just a thought, and is not something which is blocking this PR. It can be done afterwards just as well if we decide to do so. WDYT?